src/latch_cli/centromere/ast_parsing.py

```diff
@@ -0,0 +1,126 @@
+import ast
+from dataclasses import dataclass
+from pathlib import Path
+from textwrap import dedent
+from typing import Literal, Optional
+
+import click
+
+
+@dataclass
+class FlyteObject:
+    type: Literal["task", "workflow"]
+    name: str
+    dockerfile: Optional[Path] = None
+
+
+def is_task_decorator(decorator_name: str) -> bool:
+    return decorator_name in {
+        # og
+        "small_task",
+        "medium_task",
+        "large_task",
+        # og gpu
+        "small_gpu_task",
+        "large_gpu_task",
+        # custom
+        "custom_task",
+        "custom_memory_optimized_task",
+        # nf
+        "nextflow_runtime_task",
+        # l40s gpu
+        "g6e_xlarge_task",
+        "g6e_2xlarge_task",
+        "g6e_4xlarge_task",
+        "g6e_8xlarge_task",
+        "g6e_12xlarge_task",
+        "g6e_16xlarge_task",
+        "g6e_24xlarge_task",
+        # v100 gpu
+        "v100_x1_task",
+        "v100_x4_task",
+        "v100_x8_task",
+    }
+
+
+class Visitor(ast.NodeVisitor):
+    def __init__(self, file: Path):
+        self.file = file
+        self.flyte_objects: list[FlyteObject] = []
+
+    def visit_FunctionDef(self, node: ast.FunctionDef):
+        if len(node.decorator_list) == 0:
+            return self.generic_visit(node)
+
+        for decorator in node.decorator_list:
+            if isinstance(decorator, ast.Name):
+                if decorator.id == "workflow":
+                    self.flyte_objects.append(FlyteObject("workflow", node.name))
+                elif is_task_decorator(decorator.id):
+                    self.flyte_objects.append(FlyteObject("task", node.name))
+
+            elif isinstance(decorator, ast.Call):
+                func = decorator.func
+                assert isinstance(func, ast.Name)
```

Comment:
might want to handle `ast.Attribute` here if people import qualified e.g.
```py
from latch import tasks

@tasks.small_task
def f():
	....
```

Reply:
ideally we would also try to resolve to a fully qualified name (i.e. only match `latch.tasks.small_task` vs any decorator that happens to share that name)

I'd leave a todo

Reply:
added todo